packages/radix-vue/src/Avatar/AvatarImage.vue

```diff
@@ -6,34 +6,61 @@ export interface AvatarImageProps extends PrimitiveProps {}
 </script>
 
 <script setup lang="ts">
-import { inject, useAttrs, watch } from 'vue'
+import { inject, onMounted, onUnmounted, onUpdated, ref, useAttrs, watch } from 'vue'
 import { Primitive, type PrimitiveProps } from '../Primitive'
 import { AVATAR_INJECTION_KEY } from './AvatarRoot.vue'
-import { type ImageLoadingStatus, useImageLoadingStatus } from './utils'
+import { type ImageLoadingStatus } from './utils'
 
 const props = withDefaults(defineProps<AvatarImageProps>(), { as: 'img' })
 
 const emits = defineEmits<AvatarImageEmits>()
 
 const injectedValue = inject(AVATAR_INJECTION_KEY)
 
-const src = useAttrs().src as string
-const imageLoadingStatus = useImageLoadingStatus(src)
+const isMounted = ref(false)
+const src = ref('')
+
+function updateStatus(status: ImageLoadingStatus) {
+  return () => {
+    emits('loadingStatusChange', status)
+    if (isMounted.value)
+      injectedValue!.imageLoadingStatus.value = status
+  }
+}
+
+onMounted(() => {
+  src.value = useAttrs().src as string
+  isMounted.value = true
+})
+
+onUnmounted(() => {
+  isMounted.value = false
+})
+
+onUpdated(() => {
+  src.value = useAttrs().src as string
+})
 
 watch(
-  imageLoadingStatus,
-  (newValue) => {
-    emits('loadingStatusChange', newValue)
-    if (newValue !== 'idle')
-      injectedValue!.imageLoadingStatus.value = newValue
+  src,
+  (newSrc) => {
+    if (!newSrc) {
+      updateStatus('error')()
+    }
+    else {
+      const image = new window.Image()
+      updateStatus('loading')()
+      image.onload = updateStatus('loaded')
+      image.onerror = updateStatus('error')
+      image.src = newSrc
```

Comment:
can we remain the `useImageLoadingStatus` composables, and make it accept a reactive `src`. Also.. there's some missing status what you changes here.